### PR TITLE
[core] #507 Replaces sc.binaryFiles by sc.wholeTextFiles

### LINKED ISSUE #507: Replace sc.binaryFiles by sc.wholeTextFiles
The API doc states that sc.binaryFiles causes performance problems when trying to read files with varied sizes. Also this API is marked experimental.

### PR DESCRIPTION
**Pull request by keerath**
fixes #507
closes #508

### CODE REVIEW

**Comment by keerath** [comment]
@amit-shar PTAL..!!

File: core/src/main/scala/com/kodebeagle/spark/job/ConsolidatedIndexJob.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  def main(args: Array[String]): Unit = {
+    val batch = args(0)
+    val conf = new SparkConf()
+      .setMaster(KodeBeagleConfig.sparkMaster)
+      .setAppName("ConsolidatedIndexJob")
+    val sc = createSparkContext(conf)
+    val rdd = makeRDD(sc, batch)
+    val repoIndex = createRepoIndex(rdd, batch)
+    val (javaRDD, scalaRDD) = (rdd.filter { case (_, file) => file._1.endsWith(".java") },
+      rdd.filter { case (_, file) => file._1.endsWith(".java") })
```
Comment (amit-shar) [inline core/src/main/scala/com/kodebeagle/spark/job/ConsolidatedIndexJob.scala]: I think you should check scala files as well. 
e.g
file._1.endsWith(".scala")

File: core/src/main/scala/com/kodebeagle/indexer/FileMetaDataIndexer.scala [hunk 2]
```diff
@@ -45,45 +53,40 @@ case class FileMetaData(repoId: Long, fileName: String, superTypes: SuperTypes,
...
-          } else {
-            log.info("Unable to create AST for file " + source.fileName +
-              "and file contents are \n" + source.fileContent)
-            None
-          }
-        }
-      filesMetaData.filter(_.isDefined).map(_.get)
+  def generateMetaData(source: SourceFile): Set[FileMetaData] = {
+    val indexEntry = mutable.Set[FileMetaData]()
+    val parser = new JavaASTParser(true)
```
Comment (amit-shar) [inline core/src/main/scala/com/kodebeagle/indexer/FileMetaDataIndexer.scala]: move this outside the function.

File: core/src/main/scala/com/kodebeagle/indexer/FileMetaDataIndexer.scala [hunk 3]
```diff
@@ -45,45 +53,40 @@ case class FileMetaData(repoId: Long, fileName: String, superTypes: SuperTypes,
...
+        getTypeLocationMethodList(unit, resolver.getMethodInvoks, idVsExternalRefs)
+      // method definition in that class
+      val methodDefinitionList = getMethodDefinitionList(unit, resolver)
+      // internal references
+      val internalRefsList = getInternalRefs(unit, resolver)
+      val fileTypes = getFileTypes(unit, resolver)
+      val superTypes = SuperTypes(resolver.getSuperType, resolver.getInterfaces.toList)
+      indexEntry += FileMetaData(source.repoId, source.fileName, superTypes, fileTypes.toList,
+        externalRefsList.toList, typeLocationVarList.toList ++ importLocationList.toList,
+        typeLocationMethodList.toList, methodDefinitionList.toList, internalRefsList.toList)
```
Comment (amit-shar) [inline core/src/main/scala/com/kodebeagle/indexer/FileMetaDataIndexer.scala]: retain the logging.

File: core/src/main/scala/com/kodebeagle/spark/job/ConsolidatedIndexJob.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  val TYPEREFS = "typereferences"
+
+  def main(args: Array[String]): Unit = {
+    val batch = args(0)
+    val conf = new SparkConf()
+      .setMaster(KodeBeagleConfig.sparkMaster)
+      .setAppName("ConsolidatedIndexJob")
+    val sc = createSparkContext(conf)
+    val rdd = makeRDD(sc, batch)
+    val repoIndex = createRepoIndex(rdd, batch)
```
Comment (amit-shar) [inline core/src/main/scala/com/kodebeagle/spark/job/ConsolidatedIndexJob.scala]: Broadcast repoIndex.

**Comment by amit-shar** [comment]
LGTM !!

**Pull request merged**